extensions/csv/src/main/java/io/deephaven/csv/reading/CsvReader.java

```diff
@@ -0,0 +1,302 @@
+package io.deephaven.csv.reading;
+
+import io.deephaven.csv.densestorage.DenseStorageReader;
+import io.deephaven.csv.densestorage.DenseStorageWriter;
+import io.deephaven.csv.parsers.ParserBase;
+import io.deephaven.csv.sinks.Sink;
+import io.deephaven.csv.parsers.context.SentinelConfiguration;
+import io.deephaven.csv.parsers.Parsers;
+import io.deephaven.csv.sinks.SinkFactory;
+import io.deephaven.csv.containers.CharSlice;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.util.*;
+import java.util.concurrent.*;
+
+public class CsvReader {
+    private final Reader reader;
+    private final SinkFactory sinkFactory;
+    /**
+     * Whether to trim leading and trailing blanks from non-quoted values.
+     */
+    private boolean ignoreSurroundingSpaces = false;
+    /**
+     * Whether to trim leading and trailing blanks from inside quoted values.
+     */
+    private boolean trim = false;
+    /**
+     * Whether the incoming data has column headers.
+     */
+    private boolean hasHeaders = true;
+    /**
+     * The quote character (used when you want field or line delimiters to be interpreted as literal text.
+     */
+    private char quoteChar = '#';
+    /**
+     * The field delimiter (the character that separates one column from the nextd0.
+     */
+    private char fieldDelimiter = ',';
+    /**
+     * Whether to run asynchronously. In particular, the operation of reading the raw file, breaking it into columns,
+     * and storing that column text in memory can run in parallel with parsing the data for a given column, and all the
+     * column parsers can themselves run in parallel.
+     */
+    private boolean async = false;
+    /**
+     * The user-defined set of parsers that participate in type inference. Defaults to ZamboniParsers.DEFAULT
+     */
+    private Set<ParserBase<?>> inference = new HashSet<>(Parsers.DEFAULT);
+    /**
+     * Client-specified headers that can be used to override the existing headers in the input (if hasHeaders is true),
+     * or to provide absent headers (if hasHeaders is false).
+     */
+    private List<String> clientSpecifiedHeaders = new ArrayList<>();
+    /**
+     * Used to force a specific parser for a specific column.
+     */
+    private Map<String, ParserBase<?>> parsers = new HashMap<>();
+    /**
+     * The parser
+     */
+    private ParserBase<?> nullParser;
```

Comment:
Is a bit confusing that the comment says "The parser" and the name of the data member is `nullParser` ?